File: src/main/java/com/aliiensmp/aliienResize/Economy/CurrencyProvider.java
package com.aliiensmp.aliienResize.Economy;

import org.bukkit.entity.Player;

/**
 * Defines a withdraw-only currency hook used for color purchases.
 */
public interface CurrencyProvider {

    /**
     * @return {@code true} if the economy plugin is enabled, {@code false} otherwise
     */
    boolean isValid();

    /**
     * @param player player to check the balance
     * @param amount amount to check in the player's balance
     * @return {@code true} if {@code player} has {@code amount} in their balance, {@code false} otherwise
     */
    boolean hasBalance(Player player, double amount);

    /**
     * @param player player to withdraw the money
     * @param amount money to withdraw
     * @return {@code true} if the transaction is successful, {@code}
     * @requires hasBalance(player, amount)
     */
    boolean withdraw(Player player, double amount);
}

File: src/main/kotlin/com/aliiensmp/aliienResize/Economy/CurrencyProvider.kt
package com.aliiensmp.aliienResize.Economy

import org.bukkit.entity.Player

interface CurrencyProvider {

    /**
     * @return `true` if the economy plugin is enabled, `false` otherwise
     */
    val isValid: Boolean

    /**
     * @return the suffix used to display prices in names/lores
     */
    val suffix: String

    /**
     * @param player player to check the balance
     * @param amount amount to check in the player's balance
     * @return `true` if [player] has [amount] in their balance, `false` otherwise
     */
    fun hasBalance(player: Player, amount: Double): Boolean

    /**
     * @param player player to withdraw the money
     * @param amount money to withdraw
     * @return `true` if the transaction is successful
     * * Requires hasBalance(player, amount)
     */
    fun withdraw(player: Player, amount: Double): Boolean
}